# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.h
# Commit: 405100db6c45
# Full Hash: 405100db6c45f0463286067df5b491f74790d9aa
# Author: Kashav Madan <kmadan@mozilla.com>
# Date: 2019-07-11 09:51:12
# Regressor Bug: 1543251
# File Overlap Count: 1
# Description:
#   Bug 1543251 - Move hasBeforeUnload from PBrowser to PWindowGlobal, r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D37003
# ==============================================================================

diff -r 0c408343080d -r 405100db6c45 dom/ipc/WindowGlobalParent.h
--- a/dom/ipc/WindowGlobalParent.h	Thu Jul 04 07:41:57 2019 +0000
+++ b/dom/ipc/WindowGlobalParent.h	Wed Jul 10 21:13:44 2019 +0000
@@ -109,6 +109,8 @@
 
   bool IsInitialDocument() { return mIsInitialDocument; }
 
+  bool HasBeforeUnload() { return mHasBeforeUnload; }
+
   already_AddRefed<Promise> ChangeFrameRemoteness(dom::BrowsingContext* aBc,
                                                   const nsAString& aRemoteType,
                                                   uint64_t aPendingSwitchId,
@@ -142,6 +144,7 @@
     mIsInitialDocument = aIsInitialDocument;
     return IPC_OK();
   }
+  mozilla::ipc::IPCResult RecvSetHasBeforeUnload(bool aHasBeforeUnload);
   mozilla::ipc::IPCResult RecvBecomeCurrentWindowGlobal();
   mozilla::ipc::IPCResult RecvDestroy();
   mozilla::ipc::IPCResult RecvRawMessage(const JSWindowActorMessageMeta& aMeta,
@@ -170,6 +173,9 @@
   bool mInProcess;
   bool mIPCClosed;
   bool mIsInitialDocument;
+
+  // True if this window has a "beforeunload" event listener.
+  bool mHasBeforeUnload;
 };
 
 }  // namespace dom
